packages/lib/src/vis/surface/hooks.ts

```diff
@@ -0,0 +1,60 @@
+import type { NumArray } from '@h5web/shared';
+import { getDims } from '@h5web/shared';
+import type { NdArray } from 'ndarray';
+import { useMemo } from 'react';
+import { Float32BufferAttribute, Uint8BufferAttribute } from 'three';
+
+export function useBufferAttributes(
+  dataArray: NdArray<NumArray>,
+  dataToColorScale: (val: number) => [number, number, number]
+) {
+  const { position, color } = useMemo(() => {
+    const { rows, cols } = getDims(dataArray);
+
+    const positions: number[] = [];
+    const colors: number[] = [];
+
+    for (let row = 0; row < rows; row++) {
+      for (let col = 0; col < cols; col++) {
+        const value = dataArray.get(row, col);
+        positions.push(col, row, value);
+        colors.push(...dataToColorScale(value));
+      }
+    }
+
+    return {
+      position: new Float32BufferAttribute(positions, 3),
+      color: new Uint8BufferAttribute(colors, 3, true),
```

Comment:
Quite similar to [`useBufferAttributes`](https://github.com/silx-kit/h5web/blob/d8a5bafec2e3f788592d5c3b417b42b9cdb8dcb3/packages/lib/src/vis/scatter/hooks.ts#L8) of the `ScatterVis`.

Note that here, I choose to create the `BufferAttribute` in the hooks. This made more sense to me and I may change the `ScatterVis` version as well to follow this.

Reply:
I agree. Not sure about the `positions` and `colors` being plain arrays, though. I mean, it probably doesn't make a huge difference, but initialising a buffer and setting values to it is most likely more memory efficient than pushing values to normal arrays.

Reply:
True, it is done a bit better in the `ScatterVis` so I think I will revisit this in a future PR and try to take the better of both worlds